include/arch/arc/arch.h

```diff
@@ -79,87 +80,70 @@ extern "C" {
 		1 << (31 - __builtin_clz(x) + 1) : \
 		1 << (31 - __builtin_clz(x)))
 
-#if defined(CONFIG_USERSPACE)
 
+#if defined(CONFIG_USERSPACE)
 #define Z_ARCH_THREAD_STACK_RESERVED \
 	(STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE)
+#else
+#define Z_ARCH_THREAD_STACK_RESERVED (STACK_GUARD_SIZE)
+#endif
 
-#if CONFIG_ARC_MPU_VER == 2
 
-#define Z_ARCH_THREAD_STACK_DEFINE(sym, size) \
-	struct _k_thread_stack_element __noinit \
-		__aligned(POW2_CEIL(STACK_SIZE_ALIGN(size))) \
-		sym[POW2_CEIL(STACK_SIZE_ALIGN(size)) + \
-		+  STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE]
+/* MPUv2 requires size must be power of 2 and >= 2048 */
+#define Z_ARCH_THREAD_STACK_SIZE(size) POW2_CEIL(STACK_SIZE_ALIGN(size))
```

Comment:
Confusing name of macro. This only applies to MPUv2, and the macro is only used in MPUv2 stack settings. Suggest to rename into something like Z_ARCH_THREAD_STACK_SIZE_PWR2 or including _MPUv2_

Reply:
done

Reply:
Z_ARCH_ namespace is reserved for kernel <--> arch API.
Since this is ARC-specific and only used by ARC internally, call it something else.

Reply:
already changed